sky/resources.py

```diff
@@ -1162,6 +1163,13 @@ def _try_validate_image_id(self) -> None:
         Raises:
             ValueError: if the attribute is invalid.
         """
+
+        if self._image_id is None and isinstance(self._cloud, clouds.GCP):
+            if self._network_tier == resources_utils.NetworkTier.BEST:
```

Comment:
```suggestion
        # When the network tier is specified to be best for GCP, the image id should be set to
        # be a special one to ensure the environment is correctly configured.
        if (self._network_tier == resources_utils.NetworkTier.BEST and self._image_id is None and isinstance(self._cloud, clouds.GCP)):
```
minor: merge the two if : )